script.js

```diff
@@ -0,0 +1,150 @@
+const finishedList = document.querySelector(".finishedList");
+const lenTodo = document.getElementById("todo-count");
+const lenFin = document.getElementById("fin-count");
+
+const TODODATA = "todoData"
+const FINDATA = "FinData"
+
+let toDoArr = [];
+let finArr = [];
+
+function loadDATA(){
+    const loadedTODO = localStorage.getItem(TODODATA);
+    const loadedFIN = localStorage.getItem(FINDATA);
+    
+    if (loadedTODO !== null) {
+        const parsedTODO = JSON.parse(loadedTODO); // parse, 가져온 것을 자바스크립트 object로 변환시켜줌
+        parsedTODO.forEach(function(input) {
+            printData(input.text,parseInt(input.id),true)
+        });
+    }
+    if (loadedFIN !== null) {
+        const parsedFIN = JSON.parse(loadedFIN); // parse, 가져온 것을 자바스크립트 object로 변환시켜줌
+        parsedFIN.forEach(function(input) {
+            printData(input.text,parseInt(input.id),false)
+        });
+    }
+}
+
+function saveDATA(){
+    localStorage.setItem(TODODATA, JSON.stringify(toDoArr));
+    localStorage.setItem(FINDATA, JSON.stringify(finArr));
+}
+
+function setLength(){
+    lenTodo.innerText = toDoArr.length
+    lenFin.innerText = finArr.length
+}
+
+function deleteData(arr,target){
+    // filter target from arr
+    // target.id가 string으로 가져와짐 - 그래서 처음에 값을 비교를 못했음
+    const result = arr.filter(element => element.id!==target.id)
+    return result
+}
+
+function deleteHandler(event){
+    // event.target의 class를 보고 todo 인지 finished인지 판별 가능
+    const target = event.target.parentNode.parentNode
+    if(target.className === "todo"){
+        // delete from todoArr
+        toDoArr = toDoArr.filter(element => element.id !== target.id)
+        // remove from parent
+        toDoList.removeChild(target);
+    }else{
+        // delete from finArr
+        finArr = finArr.filter(element => element.id !== target.id)
+        finishedList.removeChild(target);
+    }
+    setLength();
+    console.log(toDoArr);
+    console.log(finArr);
+    saveDATA();
+}
+
+function moveData(event){
+    const target = event.target.parentNode
+    if(target.className === "todo"){
+        //move to bottom
+        target.classList.remove("todo");
+        target.classList.add("fin");
+        finArr = [...finArr,{id:target.id,text:target.firstChild.innerHTML}]
+        toDoArr = deleteData(toDoArr,target)
+
+        finishedList.appendChild(target)
+    }else{
+        //move to top
+        target.classList.remove("fin");
+        target.classList.add("todo");
+        toDoArr= [...toDoArr,{id:target.id,text:target.firstChild.innerHTML}]
+        finArr = deleteData(finArr,target)
+
+        toDoList.appendChild(target)
+    }
+    setLength();
+    saveDATA();
+}
+
+function printData(text,id,field) {
+    // 새로운 toDo 리스트 생성 함수
+    const toDo = document.createElement("li");
+    const deleteBtn = document.createElement("button");
+    const toDoContent = document.createElement("span");
+
+    //field에 따라 분기
+    if(field === true){
```

Comment:
field 라는 argument 이름으로는 이것이 어떤 용도인지 단박에 알기는 살짝 힘들 것 같다는 생각이 듭니다!
이런 식으로 field 여부에 따라 true/false를 구분해주는 함수를 만들어서 사용해도 좋을 것 같아요!
```suggestion
    const isFinishedItem = () => { return field? false : true; }
    if(!isFinishedItem){
```

Reply:
오... 매번 flag 세워서 분기처리만 했지, 이런 방법으로 코딩해 본 적은 없는 것 같아요
또 새로운 방법을 배워갑니다!